pycodestyle.py

```diff
@@ -1576,7 +1580,8 @@ def ambiguous_identifier(logical_line, tokens):
     brace_depth = 0
     idents_to_avoid = ('l', 'O', 'I')
     prev_type, prev_text, prev_start, prev_end, __ = tokens[0]
-    for index in range(1, len(tokens)):
+    len_tokens = len(tokens)
```

Comment:
```
❯ python -m timeit -s 'l = [0] * 10000' -- <<EOF
> if len(l):
>   if len(l):
>     pass
> EOF
20000000 loops, best of 5: 6.54 nsec per loop
```

And 

```
❯ python -m timeit -s 'l = [0] * 10000' -- 'len(l)'
5000000 loops, best of 5: 20.9 nsec per loop
```

This is not even a microoptimization. `len()` is not expensive not for the number of tokens we're looking at. Even if I make the list significantly larger:

```
❯ python -m timeit -s 'l = [0] * 1000000000' -- <<EOF
if len(l):
  if len(l):
    pass
EOF
50000000 loops, best of 5: 7.08 nsec per loop
```

It doesn't make that significant of a difference.

Reply:
My point is that a `micro` optimization might be tiny but still impactful. This is an increase in cognitive complexity without meaningful impact it seems.